src/main/java/EnergyDetermination.java

```diff
@@ -10,45 +12,144 @@
 
 public class EnergyDetermination
```

Comment:
Seems like there is a lot of duplicate code in these methods. Can you reduce the it?